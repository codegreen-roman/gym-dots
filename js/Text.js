import React from 'react'
import { Translate } from 'react-redux-i18n'
import { string } from 'prop-types'

const Text = ({ text }) => 
    <Translate value={text} />

Text.propTypes = {
    text: string.isRequired
}

export default Text