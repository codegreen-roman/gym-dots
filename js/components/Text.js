import React from 'react'
import { Translate } from 'react-redux-i18n'
import { string } from 'prop-types'

export const Text = ({ text }) =>
    <Translate value={text} />

Text.propTypes = {
    text: string.isRequired
}
