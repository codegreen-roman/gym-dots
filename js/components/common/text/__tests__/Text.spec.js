import React from 'react'
import { shallow } from 'enzyme'
import { Text } from '../Text'

describe('Text component', () => {
    it('should render defaults', () => {
        const wrapper = shallow(<Text text='greeting' />)
        expect(wrapper).toMatchSnapshot()
    })
})
