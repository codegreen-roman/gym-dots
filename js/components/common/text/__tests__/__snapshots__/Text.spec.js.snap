// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component should render \`ee\` version of greeting 1`] = `"Tere maailm!"`;

exports[`Text component should render \`ru\` version of greeting 1`] = `"Привет мир!"`;

exports[`Text component should render defaults, which is \`en\` version of greeting 1`] = `"Hello World!"`;
